File: sample/src/main/java/com/mandria/android/mvp/example/di/modules/AppModule.java
package com.mandria.android.mvp.example.di.modules;

import com.mandria.android.mvp.example.managers.TaskManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by michael on 19/04/2017.
 */
@Module
public class AppModule {

    @Singleton
    @Provides
    public TaskManager getTaskManager() {
        return new TaskManager();
    }
}

File: demo_dagger_kotlin/src/main/kotlin/com/mandria/android/mvp/kotlin/example/di/AppModule.kt
package com.mandria.android.mvp.kotlin.example.di

import com.mandria.android.mvp.kotlin.example.managers.TaskManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

    @Singleton
    @Provides
    fun getTaskManager(): TaskManager = TaskManager()
}